### PR TITLE
Add a way to track the ConfiguredGame of a GameWorld.

### PR DESCRIPTION
**Pull request by arthurbambou**

### CODE REVIEW

File: src/main/java/xyz/nucleoid/plasmid/game/config/GameConfigs.java [hunk 1]
```diff
@@ -80,6 +80,15 @@ private static Identifier identifierFromPath(Identifier location) {
         return CONFIGURED_GAMES.get(identifier);
     }
 
+    @Nullable
+    public static Identifier getID(ConfiguredGame<?> configuredGame) {
```
Comment (SuperCoder7979) [inline src/main/java/xyz/nucleoid/plasmid/game/config/GameConfigs.java:84]:
```suggestion
    public static Identifier getId(ConfiguredGame<?> configuredGame) {
```

File: src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java [hunk 2]
```diff
@@ -4,23 +4,31 @@
...
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 public final class ConfiguredGame<C> {
     public static final Codec<ConfiguredGame<?>> CODEC = GameType.REGISTRY.dispatchStable(c -> c.type, ConfiguredGame::codecFor);
 
     private final GameType<C> type;
     private final C config;
+    private static List<ConfiguredGame<?>> openingGames = new ArrayList<>();
```
Comment (SuperCoder7979) [inline src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: Is there a reason why this is a list? A non-list based approach may be easier here 😅

Comment (arthurbambou) [reply src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: If two game opens before the GameWorld is open, the ConfiguredGame will not match the GameWorld.

Comment (SuperCoder7979) [reply src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: That's true- but a list doesn't seem like the most appropriate solution in this instance.

Comment (arthurbambou) [reply src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: What would you suggest then?

Comment (SuperCoder7979) [reply src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: A usecase for this feature would probably help in making that decision 😅

Comment (arthurbambou) [reply src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: I wanted to show more information in game-gui when choosing a game to join. And to get those info I need to be able to know the ConfiguratedGame being played in a GameWorld.

Comment (arthurbambou) [reply src/main/java/xyz/nucleoid/plasmid/game/ConfiguredGame.java:16]: What it looks like rn. image

**Comment by gegy1000** [comment]
I think the approach to take here is pass the game initializers an object that creates GameWorlds under the specific config, so the game initializer never has to know what config is running it

**Comment by arthurbambou** [comment]
> so the game initializer never has to know what config is running it
What do you mean?

**Comment by gegy1000** [comment]
Sorry, didn't explain so great. Just pushed a commit that implements it: https://github.com/NucleoidMC/plasmid/commit/e5758dc797fc2143a6edc281e57fcfd3759dac12

**Comment by arthurbambou** [comment]
awesome! I gonna close this then.

**Pull request closed without merging**